enhancements/network/on-prem-bgp-networking.md

```diff
@@ -0,0 +1,447 @@
+depends on the upstream router and BGP infrastructure that is in place, and not managed by OpenShift.
+
+
+### baremetal-runtimecfg
+
+[baremetal-runtimecfg](https://github.com/openshift/baremetal-runtimecfg) is used for rendering the manifests and templates of the machine-config-operator project, it also supports runtime update of the configuration templates (e.g: update FRR config in case a new BGP peer added to the Failure Domain) based on the current system status.
+
+The next capabilities are supported by the `baremetal-runtimecfg`:
+- `renders templates` using [values provided at command line](https://github.com/openshift/baremetal-runtimecfg/blob/master/cmd/runtimecfg/render.go), parameters retrieved from the cluster (e.g: control plane nodes) and
+  - api-vips
+  - ingress-vips
+- `frr monitor`: verify that the desired configuration is synced with the cluster. As seen before, it's used from a side-car container in the FRR pod.
+
+### API Extensions
+
+None.
+
+### Implementation Details/Notes/Constraints
+
+#### MetalLB
+
+MetalLB is a load-balancer implementation for bare metal Kubernetes clusters, using standard routing protocols and backed by FRR when using BGP as a protocol.
+
+For now, we recommend separating the MetalLB deployment from the workers where Ingress Controller runs.
+This is because the MetalLB deployment is configured to use the same BGP peers as the ones used by our implementation.
+The [MetalLB documentation](https://metallb.universe.tf/configuration/_advanced_bgp_configuration/#limiting-peers-to-certain-nodes) explains how to decide where to run peers on certain nodes by using the node-selectors attribute of peers in the Peers configuration. The semantics of these selectors are the same as those used elsewhere in Kubernetes, so refer to the [labels documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels).
+
+#### Customer requirements
+
+With this new implementation, the customer needs to have a BGP infrastructure in place to be able to deploy and use the OpenShift cluster (which is why we will never make it the default implementation).
+
+Here is a list of requirements that the customer needs to have in place:
+
+- A BGP router (or multiple) that can peer with the BGP peers on the control plane nodes.
+- ICMP redirects are disabled on the router(s) that will peer with the control plane nodes. This is needed to ensure that the router(s) does not send an ICMP redirect message when traffic destined to the VIPs is received on a node where the VIP is not configured. This is needed to ensure that the traffic is not black-holed in the network.
+
+We will need to document the requirements and the steps to configure the BGP infrastructure, as well as providing as much as possible some validation tools to help the customer to configure the BGP infrastructure correctly.
```

Comment:
See also LB options, e.g. ensuring the hash algorithm uses source port, and ideally is also consistent.

Reply:
I added a third requirement taking in account your suggestion.